addons/utm/models/utm_mixin.py

```diff
@@ -59,3 +55,29 @@ def tracking_fields(self):
             ('utm_source', 'source_id', 'odoo_utm_source'),
             ('utm_medium', 'medium_id', 'odoo_utm_medium'),
         ]
+
+    def _find_or_create_record(self, model_name, name):
+        """Based on the URL parameter, retrieve the corresponding record or create it."""
+        Model = self.env[model_name]
+
+        record = Model.search([('name', 'ilike', name)], limit=1)
+
+        if not record:
+            # No record found, create a new one
+            record_values = {'name': name}
+            if 'is_auto_campaign' in record._fields:
+                record_values['is_auto_campaign'] = True
+            record = Model.create(record_values)
+
+        return record
+
+    @api.model
+    def _generate_unique_name(self, record, name):
+        """Try to use the given name for the given model.
+
+        If the name already exists, add random characters at the end.
+        """
+        same_name = self.env[record._name].search_count([('name', '=', name)])
+        if same_name:
+            name = f'{name} [{str(uuid.uuid4())[:8]}]'
```

Comment:
Strange, why not a counter or something like that ? 8 random characters is a bit out of the blue, isn't it ?

Reply:
Not sure to see what you mean ... if we don't play with UUID, why would we have to remove the potential UUID part ?